### ISSUE TITLE
Gem vs Engine user interface clash.

### ISSUE THREAD

**Issue by safarista** [rubysherpas/forem#178]
Ok. I was tweeting about with Ryan Bigg about the STYLE in which Forem is implemented. Personally I have issues with the way Engines work. Some gems are implemented like Engines and it gets on my b****.

Looking through authentication gems I prefer using [sorcery](https://github.com/NoamB/sorcery) because of the way its implemented. I rather prefer implementing my own user auth than having to customise 60% of the UI that a gem provides. I will quote NoamB on this.
### Philosophy

Sorcery is a stripped-down, bare-bones authentication library, with which you can write your own authentication flow. It was built with a few goals in mind:
- Less is more - less than 20 public methods to remember for the entire feature-set make the lib easy to ‘get’.
- No built-in or generated code - use the library’s methods inside _your own_ MVC structures, and don’t fight to fix someone else’s.
- Magic yes, Voodoo no - the lib should be easy to hack for most developers.
- Configuration over Confusion - Centralized (1 file), Simple & short configuration as possible, not drowning in syntactic sugar.
- Keep MVC cleanly separated - DB is for models, sessions are for controllers. Models stay unaware of sessions.

So in short its the style and weight I have issues with. I would prefer that the gem provides the helper methods and stay out of the way of the User Interface.

Just my preference.

**Comment by radar** [comment]
I still don't understand this. Forem doesn't interfere with the User Interface in any other way besides adding a necessary `Forem::DefaultPermissions` to the user class configured with the `Forem.user_class`.

What specific issue are you encountering?

**Comment by safarista** [comment]
Its not the Authentication. Its the Interface. I should probably read the theme-ing process properly. But the features are kind of hidden. I'll do the digging myself. Disregard my issue.

**Comment by radar** [comment]
Please let us know if you have a question that isn't answered on the wiki page: https://github.com/radar/forem/wiki/Theming.

**Comment by safarista** [comment]
I added another engine on my app, broke the routes generation. `The main_app.*` helper is conflicting and complicating my routes. So is my concern above making any sense?

**Comment by radar** [comment]
Conflicting how? This helper should not conflict with anything.

**Comment by murtazajafferji** [comment]
I am using Spree. I think it is also interfering with my routes.
